### PR TITLE
Fix hangs when executing pytest

### PR DESCRIPTION
**Pull request by MagicPro1994**
I solved my hang problem at the end of pytest execution by add timeout to workerpool.waitall.
I hope this will help you somehow.

### CODE REVIEW

File: execnet/multi.py [hunk 1]
```diff
@@ -295,7 +295,7 @@ def termkill(termfunc, killfunc):
         replylist.append(reply)
     for reply in replylist:
         reply.get()
-    workerpool.waitall()
+    workerpool.waitall(timeout=timeout)
```
Comment (MagicPro1994) [inline execnet/multi.py]:
```suggestion
    workerpool.waitall(timeout)
```

**Comment by nicoddemus** [comment]
@hpk42 any chance of you taking a look at this? I've zero knowledge of execnet's internals to know the ramifications of this change.

**Comment by hpk42** [comment]
tests fail on python2.7 but i think we could remove py27 support sometime soon anyway.  maybe removing py27 from CI-testing makes sense.

**Comment by hpk42** [comment]
otherwise i think it's fine to merge this if it helps real-world problems ;)

**Comment by nicoddemus** [comment]
1.7.0 released, thanks and sorry for the delay!

**Pull request merged**